### PR TITLE
rifiuti: init at 20040505_1

### PR DESCRIPTION
**Pull request by D3vil0p3r**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ri/rifiuti/package.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  src = fetchzip {
+    url = "mirror://sourceforge/project/odessa/Rifiuti/${finalAttrs.version}/rifiuti_${finalAttrs.version}.zip";
+    hash = "sha256-bVPz0nXexGtQtXxGb3Mc79tzyZikc7KxNNWuvgu6pQ0=";
+  };
+
+  patches = [
+    (fetchpatch {
+      url = "https://salsa.debian.org/pkg-security-team/rifiuti/-/raw/f237358a91b12776beb9942c79ccb3aea180968a/debian/patches/add-GCC-hardening";
+      hash = "sha256-4cQSNtnHRD88TEP2iESgFlY16Y8WZZityIQh+M1kchQ=";
+    })
```
Comment (emilytrau) [inline pkgs/by-name/ri/rifiuti/package.nix:20]: Consider adding the name attribute like so:
```suggestion
    (fetchpatch {
      name = "add-GCC-hardening.patch";
      url = "https://salsa.debian.org/pkg-security-team/rifiuti/-/raw/f237358a91b12776beb9942c79ccb3aea180968a/debian/patches/add-GCC-hardening";
      hash = "sha256-4cQSNtnHRD88TEP2iESgFlY16Y8WZZityIQh+M1kchQ=";
    })
```

File: pkgs/by-name/ri/rifiuti/package.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    (fetchpatch {
+      url = "https://salsa.debian.org/pkg-security-team/rifiuti/-/raw/5bd48acbcb63cee324556c4ec29750ce1e41755c/debian/patches/use-CC-and-abort-on-error.patch";
+      hash = "sha256-aj1ubNMzzpBAyyEN/+zcdC0oC5eYBSDte7wR8isIyMw=";
+    })
+  ];
+
+  postPatch = ''
+    substituteInPlace src/Makefile \
+      --replace gcc cc
```
Comment (emilytrau) [inline pkgs/by-name/ri/rifiuti/package.nix:37]:
```suggestion
      --replace-fail gcc cc
```

File: pkgs/by-name/ri/rifiuti/package.nix [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  makeFlags = [
+    "-C src"
+  ];
+
+  enableParallelBuilding = true;
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    cp bin/rifiuti $out/bin
```
Comment (emilytrau) [inline pkgs/by-name/ri/rifiuti/package.nix:49]:
```suggestion
    install -D bin/rifiuti $out/bin/rifiuti
```

File: pkgs/by-name/ri/rifiuti/package.nix (hunk 1 above)
Comment (emilytrau) [reply pkgs/by-name/ri/rifiuti/package.nix:20]: `fetchpatch` is only needed over `fetchurl` when the patch is dynamically generated from a source control system. This seems to be a static file so `fetchurl` should be fine.

Comment (D3vil0p3r) [reply pkgs/by-name/ri/rifiuti/package.nix]: Fixed

**Comment by emilytrau** [comment]
Result of `nixpkgs-review pr 289762` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>rifiuti</li>
  </ul>
</details>

**Review by emilytrau — APPROVED** [review]

**Pull request merged**